aiidalab_widgets_base/viewers.py

```diff
@@ -71,30 +84,182 @@ class StructureDataViewer(ipw.VBox):
     :param downloadable: If True, add link/button to download the content of the object
     :type downloadable: bool"""
 
-    def __init__(self, structure=None, downloadable=True, **kwargs):
+    SELECTION_SIZE = 10
 
+    def __init__(self, structure=None, downloadable=True, configure_view=True, **kwargs):
         self._structure = None
         self._name = None
         self._viewer = nglview.NGLWidget()
+        self._viewer.camera = 'orthographic'
+        self._viewer.observe(self._on_atom_click, names='picked')
+        self._viewer.stage.set_parameters(mouse_preset='pymol')
+
+        layout = {"description_width": "initial", "width": "95%"}
+        center_button = ipw.Button(description="Center", layout=layout)
+        center_button.on_click(lambda c: self._viewer.center())
+
+        # Choose background color.
+        background_color = ipw.ColorPicker(description="Background", layout=layout)
+
+        def change_background(change):  # Note: change this to lambda when switched to python3.8
+            self._viewer.background = change['new']
+
+        background_color.observe(change_background, names="value")
+        background_color.value = 'white'
+
+        # Make screenshot.
+        screenshot = ipw.Button(description="Screenshot")
+        screenshot.on_click(lambda c: self._viewer.download_image())
+
+        # Selected atoms.
+        self._selected_atoms = ipw.Textarea(description='Selected atoms:',
+                                            value='',
+                                            style={'description_width': 'initial'})
+        self._selected_atoms.observe(self._apply_selection, names='value')
+        self.wrong_syntax = ipw.HTML(
+            value="""<i class="fa fa-times" style="color:red;font-size:2em;" ></i> wrong syntax""",
+            layout={'visibility': 'hidden'})
+        self._selection = set()
+        copy_selection_to_clipboard = ipw.Button(description="Copy to clipboard")
+        clear_selection = ipw.Button(description="Clear selection")
+        clear_selection.on_click(self.clear_selection)
+
+        def copy_to_clipboard(change=None):  # pylint:disable=unused-argument
```

Comment:
I'd suggest to move this into a module function, potentially as part of a util module.

Reply:
ok, nice idea. Will move it to utils.

Reply:
done